Sources/Workspace/Workspace.swift

```diff
@@ -4185,6 +4189,169 @@ extension Workspace.Location {
     }
 }
 
+#warning("tomer: experiment")
+extension Workspace {
+    fileprivate struct RegistryAwareManifestLoader: ManifestLoaderProtocol {
```

Comment:
this is the key change. it basically translates source control dependencies to registries ones when possible.

this is still in wip state but the concept is clear